api/src/main/java/org/apache/iceberg/SortOrderBuilder.java

```diff
@@ -103,4 +103,12 @@ default R desc(Term term) {
    * @return this for method chaining
    */
   R desc(Term term, NullOrder nullOrder);
+
+  /**
+   * Set case sensitivity of sort column name resolution.
+   *
+   * @param caseSensitive when true, column name resolution is case-sensitive
+   * @return this for method chaining
+   */
+  R caseSensitive(boolean caseSensitive);
```

Comment:
What about throwing `UnsupportedOperationException` instead of changing API checks?

Reply:
done